src/coprocessor/checksum.rs

```diff
@@ -56,7 +56,21 @@ impl<S: Snapshot> RequestHandler for ChecksumContext<S> {
         let mut checksum = 0;
         let mut total_kvs = 0;
         let mut total_bytes = 0;
-        while let Some((k, v)) = self.scanner.next()? {
+        let (old_prefix, new_prefix) = if self.req.has_rule() {
+            let mut rule = self.req.get_rule().clone();
+            (rule.take_old_prefix(), rule.take_new_prefix())
+        } else {
+            (vec![], vec![])
+        };
+        while let Some((mut k, v)) = self.scanner.next()? {
+            // undo rewrite
+            if !old_prefix.is_empty() && !new_prefix.is_empty() {
+                assert!(k.starts_with(&new_prefix));
```

Comment:
🤔 If we give the wrong `new_prefix` it will crash TiKV?

Reply:
fixed